### PR TITLE
v1.9 backports 2021-07-05

### PR DESCRIPTION
**Pull request by twpayne**
* #16190 -- pkg/k8s: add pod IP event change (@aanm)
 * #16554 -- ci: Disable NFS locking (@gandro)
 * #16572 -- contrib: Identify upstream commits by author and date (@pchaigno)
 * #16721 -- Fix and add more commands in CI sysdumps (@aanm)
 * #16690 -- Docs: Fix maglev.hashSeed byte size documentation (@gaffneyd4)
 * #16750 -- contrib/docs: rename 'cilium-actions.yml' with 'maintainers-little-helper.yaml (@aanm)
 * #16756 -- Revert "docs: add 'endpointRoutes.enabled=true' to aws-cni" (@bmcustodio)
 * #16691 -- bugtool: Collect BPF cgroup programs related information (@aditighag)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 16190 16554 16572 16721 16690 16750 16756 16691; do contrib/backporting/set-labels.py $pr done 1.9; done
```

### CODE REVIEW

**Comment by twpayne** [comment]
The following commits required conflict resolution, please check these carefully.

| Commit in this branch | Commit in master branch | PR | Author |
| - | - | - | - |
| e92dc6ac6b766e793091410d0cf58c61b01d424d | ffd92836f77aed1512be7dda4eca4f356a8fba56 | #16190 | @aanm |
| edf76fb1ef6b58d5ef90b439d54134f314ed086e | fb723f8133c40faa068a5a401f594622668b2753 | #16589 | @pchaigno |
| f470a071bd5df373578f720f852a0bd1c53731d8 | c7f01bc7793db0cf75f89df259647df6af9ee2f8 | #16721 | @aanm |
| d936ebf18cc329628529d7881cf5c86082de3fec | 70596cbe64f7e061a9bf5303b09452238c8dd334 | #16750 | @aanm |

**Review by gandro — APPROVED** [review]
My commit looks good!

**Comment by twpayne** [comment]
@jibi #16136 is marked for backport to 1.9 but both commits in the PR result in conflicts when applied to the `v1.9` branch that I am unsure how to resolve. Could you create a PR backporting #16136 to the `v1.9` branch please?

**Comment by twpayne** [comment]
test-backport-1.9

**Review by bmcustodio — APPROVED** [review]
My changes LTGM!

File: test/k8sT/manifests/log-gatherer.yaml [hunk 1]
```diff
@@ -30,8 +30,8 @@ spec:
         - "10000"
         command:
         - sleep
-        image: docker.io/cilium/log-gatherer:v1.0
-        imagePullPolicy: Always
+        image: docker.io/cilium/log-gatherer:v1.1
```
Comment (aanm) [inline test/k8sT/manifests/log-gatherer.yaml:33]: This change is not part of the original commit. @nebril or @nbusseneau is it ok to bump the version or should we keep it to v1.0?

**Review by aanm — APPROVED** [review]

**Review by pchaigno — APPROVED** [review]
My changes look good :+1:

File: pkg/k8s/watchers/pod.go [hunk 2]
```diff
@@ -272,20 +280,19 @@ func (k *K8sWatcher) updateK8sPodV1(oldK8sPod, newK8sPod *slim_corev1.Pod) error
...
-	// Process IP updates
-	k.addK8sPodV1(newK8sPod)
+	oldPodIPs := k8sUtils.ValidIPs(oldK8sPod.Status)
+	newPodIPs := k8sUtils.ValidIPs(newK8sPod.Status)
+	err := k.updatePodHostData(oldK8sPod, newK8sPod, oldPodIPs, newPodIPs)
+
+	if err != nil {
+		logger.WithError(err).Warning("Unable to update ipcache map entry on pod update")
+		return err
+	}
```
Comment (aanm) [inline pkg/k8s/watchers/pod.go]:
```suggestion
       switch {
       case err != nil:
               msg := "Unable to update ipcache map entry on pod add"
               if err == errIPCacheOwnedByNonK8s {
                       logger.WithError(err).Debug(msg)
               } else {
                       logger.WithError(err).Warning(msg)
               }
       default:
               logger.Debug("Updated ipcache map entry on pod add")
       }
```

**Review by aanm — CHANGES_REQUESTED** [review]
Tests are failing. Please use the provided patch to fix them

File: pkg/k8s/watchers/pod.go (hunk 2 above)
Comment (aanm) [reply pkg/k8s/watchers/pod.go]: also add this in line 115
```golang
	errIPCacheOwnedByNonK8s = fmt.Errorf("ipcache entry owned by kvstore or agent")
```

**Comment by twpayne** [comment]
test-backport-1.9

File: pkg/k8s/watchers/pod.go (hunk 2 above)
Comment (twpayne) [reply pkg/k8s/watchers/pod.go]: First change made. Second change does not seem to be needed as `errIPCacheOwnedByNonK8s` is already defined at `pkg/k8s/watchers/watcher.go:110`.

**Comment by twpayne** [comment]
PR updated to remove #16387

**Comment by twpayne** [comment]
test-backport-1.9

**Comment by twpayne** [comment]
@pchaigno The Cilium-PR-K8s-Upstream test is failing with the error ([link to logs](https://jenkins.cilium.io/job/Cilium-PR-K8s-Upstream/3199/execution/node/76/log/)):

```
12:25:46  result match, continue with kubernetes
12:25:46  Installing kubetest manually
12:25:46  mkdir: cannot create directory '/home/vagrant/go/src/k8s.io': Permission denied
```

This looks like its related to #16589. Could you investigate?

**Review by aanm — APPROVED** [review]

**Comment by twpayne** [comment]
test-backport-1.9

**Review by aditighag — APPROVED** [review]
Backport for my commit looks good! Thanks!

File: test/k8sT/manifests/log-gatherer.yaml (hunk 1 above)
Comment (nebril) [reply test/k8sT/manifests/log-gatherer.yaml:33]: If the CI passes it is safe to bump the version.

**Comment by twpayne** [comment]
test-backport-1.9

**Comment by twpayne** [comment]
test-upstream-k8s

**Comment by twpayne** [comment]
k8s-1.18-kernel-4.19 failure looks like #15097, which is supposed to be fixed. I don't think the fix (9e086277de3e23450953c2afd24a68fd727d3066) was backported to 1.9 though.

k8s-1.17-kernel-5.4 failure has produced more than 2GB of logs, which means Jenkins cannot display them. Trying other methods to investigate.

**Comment by twpayne** [comment]
k8s-1.12-kernel-netnext failure looks like #12511.

k8s-1.17-kernel-4.9 failure looks like an infrastructure flake: an initial NodeJS script fails but does not produce an error message, and then multiple tests fail in a later step.

k8s-1.17-kernel-5.4 failure looks like a K8sVerifier test failure.

k8s-1.18-kernel-4.19 failure looks like #15097.

**Comment by twpayne** [comment]
test-1.12-netnext

**Comment by twpayne** [comment]
test-1.17-4.9

**Comment by twpayne** [comment]
test-1.17-5.4

**Comment by twpayne** [comment]
test-1.18-4.19

**Comment by twpayne** [comment]
k8s-1.17-kernel-5.4 failure seems to be a genuine K8sVerifier test failure. From the 2GB logs:

```
973: (b7) r9 = 60
BPF program is too large. Processed 1000001 insn
processed 1000001 insns (limit 1000000) max_states_per_insn 40 total_states 60448 peak_states 2529 mark_read 97

Error filling program arrays!
Unable to load program
command terminated with exit code 1

FAIL: Expected the kernel verifier to pass for BPF programs
```

What's strange though is that there are no changes to the datapath in this PR.

**Comment by pchaigno** [comment]
It's https://github.com/cilium/cilium/issues/16050.

**Pull request merged**